pennylane/wires.py

```diff
@@ -0,0 +1,399 @@
+            n_samples (int): number of subsampled wires
+            seed (int): optional random seed used for selecting the wires
+
+        Returns:
+            Wires: random subset of wires
+        """
+
+        if n_samples > len(self):
+            raise WireError("cannot sample {} wires from {} wires.".format(n_samples, len(self)))
+
+        if seed is not None:
+            np.random.seed(seed)
+
+        indices = np.random.choice(len(self), size=n_samples, replace=False)
+        subset = [self.wire_list[i] for i in indices]
+        return Wires(subset)
+
+    @staticmethod
+    def shared_wires(list_of_wires):
+        """Return only the wires that appear in each Wires object in the list.
+
+        This is similar to a set intersection method, but keeps the order of wires as they appear in the list.
+
+        For example:
+
+        >>> wires1 =  Wires([4, 0, 1])
+        >>> wires2 = Wires([3, 0, 4])
+        >>> wires3 = Wires([4, 0])
+        >>> Wires.shared_wires([wires1, wires2, wires3])
+        <Wires [4, 0]>
+
+        >>> Wires.shared_wires([wires2, wires1, wires3])
+        <Wires [0, 4]>
+
+        Args:
+            list_of_wires (List[Wires]): List of Wires objects
+
+        Returns:
+            Wires: shared wires
+        """
+
+        for wires in list_of_wires:
+            if not isinstance(wires, Wires):
+                raise WireError(
+                    "expected a `pennylane.wires.Wires` object; got {} of type {}".format(
+                        wires, type(wires)
+                    )
+                )
+
+        shared = []
+        # only need to iterate through the first object,
+        # since any wire not in this object will also not be shared
+        for wire in list_of_wires[0]:
+            if all(wire in wires_ for wires_ in list_of_wires):
+                shared.append(wire)
+
+        return Wires.merge(shared)
+
+    @staticmethod
+    def all_wires(list_of_wires):
+        """Return the wires that appear in any of the Wires objects in the list.
+
+        This is similar to a set combine method, but keeps the order of wires as they appear in the list.
+
+        For example:
+
+        >>> wires1 =  Wires([4, 0, 1])
+        >>> wires2 = Wires([3, 0, 4])
+        >>> wires3 = Wires([5, 3])
+        >>> list_of_wires = [wires1, wires2, wires3]
+        >>> Wires.all_wires(list_of_wires)
+        <Wires = [4, 0, 1, 3, 5]>
+
+        Args:
+            list_of_wires (List[Wires]): List of Wires objects
+
+        Returns:
+            Wires: combined wires
+        """
+
+        combined = []
+        for wires in list_of_wires:
+            if not isinstance(wires, Wires):
+                raise WireError(
+                    "expected a `pennylane.wires.Wires` object; got {} of type {}".format(
+                        wires, type(wires)
+                    )
+                )
+
+            combined.extend(wire for wire in wires.wire_list if wire not in combined)
```

Comment:
True, but it makes making a new Wires object easier...

Reply:
But with stricter input checks we can design the class so that ``Wires([Wires([0]), Wires([1])])`` produces ``<Wires = [0, 1]>``, which makes the merge method superfluous...